src/Application/Actions/Hooks/StripeChargeUpdate.php

```diff
@@ -288,7 +288,7 @@ private function handleChargeRefunded(Event $event, Response $response): Respons
                 $donation->getUuid(),
                 $charge->id,
             ));
-            $donation->setDonationStatus(DonationStatus::Refunded);
+            $donation->recordRefundAt(new \DateTimeImmutable());
```

Comment:
We probably could take all the timestamps above from stripe, but I'm not sure if there's any particular advantage for that over just taking them from the server clock. Depends how precise we need the date to be. @DomTBG do you have a view?